src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2022 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {useTable, Column} from "react-table";
+import {Spinner, SpinnerSize} from "@dellstorage/clarity-react/spinner/Spinner";
+import {Constants} from "components/DataGridWithInfiniteScroll/Constants";
+
+/**
+ * General component description :
+ * DataGridWithInfiniteScroll :
+ * Datagrids are for organizing large volumes of data that users can perform actions on.
+ */
+
+/**
+ * @param {rows} rows data
+ * @param {columns} column details
+ * @param {className} CSS class names
+ * @param {style} to add custom CSS styles
+ * @param {isLoading} if true then show loading icon before table is rendered
+ * @param {className} is to get a custom class to be applied or else default class is applied.
+ */
+type DataGridProps = {
+    rows: {[key: string]: any}[];
+    columns: Column<{[key: string]: any}>[];
+    style?: any;
+    isLoading?: boolean;
+    className?:string;
+};
+
+function DataGridWithInfiniteScroll(props: DataGridProps) {
+    const {style, isLoading, columns, className} = props;
+    const data = props.rows;
+
+    /**
+     * @param {getTableProps} is a function to resolve any props needed by the react table wrapper.
+     * @param {getTableBodyProps } to render table body of react table
+     * @param {headerGroups } return headers which is going to be the headers of the table
+     * @param {rows} display rows of the datagrid
+     * @param {prepareRow} is a function that must be called on any row to be displayed. It is responsible for preparing a row for rendering.
+     * @param {useTable } hook takes options and plugins to build a table instance. The basic options are columns and row data
+     */
+    const {getTableProps, getTableBodyProps, headerGroups, rows, prepareRow} = useTable({
+        columns,
+        data,
+    });
+
+    /**
+     * Function to render react table Header
+     * @param {getHeaderGroupProps} to map the headerGroups to show the individual <tr> by getting the getHeaderGroupProps()
```

Comment:
This comment seems misplaced.  The method `renderTableHeader` dont have any method Arguments.  Please check.

Reply:
I have removed the comment as only functional description was required.